src/modules/70-pipeline/utils/pipelineReconcileUtils.ts

```diff
@@ -42,29 +54,29 @@ function hasDiffTemplateInputsAndStageOrPipeline<T extends { [key: string]: any
 ): { hasDifference: boolean } {
   const templatePaths = getPaths(templateInputs)
   const templatePaths_RuntimeOnly = filterRuntimeInputs(templateInputs, templatePaths)
-  const templatePaths_WithoutIndex = templatePaths
-  const templatePaths_WithoutIndex_RuntimeOnly = templatePaths_RuntimeOnly
 
   const pipelinePaths = getPaths(pipelineOrStage).map(path => path.replace(new RegExp(`^${removePrefix}`), ''))
   const pipelinePaths_RuntimeOnly = filterRuntimeInputs(pipelineOrStage, pipelinePaths, removePrefix)
-  const pipelinePaths_WithoutIndex = pipelinePaths
-  const pipelinePaths_WithoutIndex_RuntimeOnly = pipelinePaths_RuntimeOnly
+  const pipelinePaths_RuntimeOnly_WithoutNested = removeNestedPaths(
+    pipelinePaths_RuntimeOnly,
+    templatePaths_RuntimeOnly
+  )
 
   let hasDifference = false
   // NOTE: When we change from runtime input to fixed value, property will be deleted. This will cause a difference in pipeline and template.
   // To prevent this, if yaml is changed (isUpdated) we are not checking if there are more fileds in template than in pipeline.
   // This means that in some edge cases algorithm will not work as expected.
   if (!isUpdated) {
-    templatePaths_WithoutIndex_RuntimeOnly.forEach((tlPath: string) => {
-      if (pipelinePaths_WithoutIndex.indexOf(tlPath) === -1) {
+    templatePaths_RuntimeOnly.forEach((tlPath: string) => {
+      if (pipelinePaths.indexOf(tlPath) === -1) {
         hasDifference = true
       }
     })
```

Comment:
Not a problem with a hotfix, but a better way than `forEach` and `indexOf` would be `some` and `includes`.
```suggestion
    if (templatePaths_RuntimeOnly.some((tlPath: string) => pipelinePaths.includes(tlPath))) {
      hasDifference = true
    }
```